From: Russell King <rmk+kernel@armlinux.org.uk>
Date: Thu, 5 Jan 2017 09:34:42 +0000
Subject: [PATCH] net: phy: allow EEE with SGMII interface modes

As EEE is able to work in SGMII mode as well, add it to the list of
permissable EEE modes that phy_init_eee() will accept.  This is
necessary so that EEE can work with an 88E1512 connected in SGMII mode.

Signed-off-by: Russell King <rmk+kernel@armlinux.org.uk>
---

--- a/drivers/net/phy/phy.c
+++ b/drivers/net/phy/phy.c
@@ -1236,6 +1236,7 @@ int phy_init_eee(struct phy_device *phyd
 	if ((phydev->duplex == DUPLEX_FULL) &&
 	    ((phydev->interface == PHY_INTERFACE_MODE_MII) ||
 	    (phydev->interface == PHY_INTERFACE_MODE_GMII) ||
+	     phydev->interface == PHY_INTERFACE_MODE_SGMII ||
 	     phy_interface_is_rgmii(phydev) ||
 	     phy_is_internal(phydev))) {
 		int eee_lp, eee_cap, eee_adv;
